Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java

```diff
@@ -73,425 +75,391 @@
+     * @param principal
+     *            {@link Principal} object to access logged in client.
+     * @return Response of XML of success in creating the concept collection in
+     *         the workspace.
+     * @throws RestException
+     * @throws QuadrigaStorageException
+     * @throws QuadrigaAccessException
+     */
+    @RestAccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.WORKSPACE_REST, paramIndex = 1, userRole = {
+            RoleNames.ROLE_WORKSPACE_COLLABORATOR_ADMIN, RoleNames.ROLE_WORKSPACE_COLLABORATOR_CONTRIBUTOR }) })
+    @RequestMapping(value = "rest/workspace/{workspaceId}/createcc", method = RequestMethod.POST)
+    public ResponseEntity<String> addConceptCollectionsToWorkspace(@PathVariable("workspaceId") String workspaceId,
+            HttpServletRequest request, HttpServletResponse response, @RequestBody String xml,
+            @RequestHeader("Accept") String accept, ModelMap model, Principal principal)
+                    throws RestException, QuadrigaStorageException, QuadrigaAccessException {
+        IUser user = usermanager.getUser(principal.getName());
+        String ccName = request.getParameter("name");
+        String desc = request.getParameter("desc");
+
+        if (!checkWSSecurity.checkIsWorkspaceExists(workspaceId)) {
+            String errorMsg = restMessage.getErrorMsg("Workspace ID : " + workspaceId + " doesn't exist", request);
+            return new ResponseEntity<String>(errorMsg, HttpStatus.NOT_FOUND);
+        }
+
+        IConceptCollection collection = conceptCollectionFactory.createConceptCollectionObject();
+
+        if (ccName == null || ccName.isEmpty()) {
+            String errorMsg = restMessage.getErrorMsg("Please provide concept collection name", request);
+            return new ResponseEntity<String>(errorMsg, HttpStatus.BAD_REQUEST);
+        }
+        if (desc == null || desc.isEmpty()) {
+            String errorMsg = restMessage.getErrorMsg("Please provide concept collection description", request);
+            return new ResponseEntity<String>(errorMsg, HttpStatus.BAD_REQUEST);
+        }
+        logger.debug("XML : " + xml);
+        JAXBElement<QuadrigaConceptReply> response1 = null;
+        try {
+            JAXBContext context = JAXBContext.newInstance(QuadrigaConceptReply.class);
+            Unmarshaller unmarshaller = context.createUnmarshaller();
+            unmarshaller.setEventHandler(new javax.xml.bind.helpers.DefaultValidationEventHandler());
+            InputStream is = new ByteArrayInputStream(xml.getBytes());
+            response1 = unmarshaller.unmarshal(new StreamSource(is), QuadrigaConceptReply.class);
+        } catch (JAXBException e) {
+            logger.error("Error in unmarshalling", e);
+            String errorMsg = restMessage.getErrorMsg("Error in unmarshalling", request);
+            return new ResponseEntity<String>(errorMsg, HttpStatus.BAD_REQUEST);
+        }
+        if (response1 == null) {

```

Comment:
according to the documentation, unmarshal never returns null. you can remove the if condition.